pkg/download/settings/download.go

```diff
@@ -191,26 +191,31 @@ func extractApiErrorMessage(err error) string {
 }
 
 func getObjectsPermission(ctx context.Context, client client.SettingsClient, objects []dtclient.DownloadSettingsObject) (map[string]dtclient.PermissionObject, error) {
-	downloadMutex := sync.Mutex{}
-	wg := sync.WaitGroup{}
-	wg.Add(len(objects))
+	type result struct {
+		Schema   dtclient.PermissionObject
+		ObjectId string
+		Err      error
+	}
 	errs := make([]error, 0)
+	resChan := make(chan result, len(objects))
+	defer close(resChan)
+
 	permissions := make(map[string]dtclient.PermissionObject)
 	for _, obj := range objects {
-		go func(obj dtclient.DownloadSettingsObject) {
-			defer wg.Done()
-
+		go func(ctx context.Context, obj dtclient.DownloadSettingsObject) {
 			permission, err := client.GetPermission(ctx, obj.ObjectId)
-			if err != nil {
-				downloadMutex.Lock()
-				errs = append(errs, err)
-				downloadMutex.Unlock()
-				return
-			}
-			permissions[obj.ObjectId] = permission
-		}(obj)
+			resChan <- result{permission, obj.ObjectId, err}
+		}(ctx, obj)
 	}
-	wg.Wait()
+
+	for i := 0; i < len(objects); i++ {
```

Comment:
Is it possible to range over the channel until it is closed? 

```suggestion
	for res := range resChan {
```

Reply:
Actually no, because we don't know when it should be closed. With a wait group we could wait for everything to be done and then close it. The implemenation was the outcome of this one https://github.com/Dynatrace/dynatrace-configuration-as-code/pull/1777#discussion_r1993449907